frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BaseResetBranchToRemoteAction.java

```diff
@@ -119,36 +126,50 @@ public void actionPerformed(AnActionEvent anActionEvent) {
         LOG.debug(() -> "Resetting '${branchName}' branch");
         try (AccessToken ignored = DvcsUtil.workingTreeChangeStarted(project, TASK_TITLE)) {
-          GitLineHandler handler = new GitLineHandler(myProject, gitRepository.getRoot(), GitCommand.RESET);
-          handler.addParameters("--keep");
+          GitLineHandler resetHandler = new GitLineHandler(myProject, gitRepository.getRoot(), GitCommand.RESET);
+          resetHandler.addParameters("--keep");
 
           Option<BaseGitMacheteBranch> branchOption = macheteRepository.getBranchByName(branchName);
           assert branchOption.isDefined() : "Can't get branch '${branchName}' from Git Machete repository";
           Option<IGitMacheteRemoteBranch> remoteTrackingBranchOption = branchOption.get().getRemoteTrackingBranch();
           if (remoteTrackingBranchOption.isDefined()) {
-            handler.addParameters(remoteTrackingBranchOption.get().getPointedCommit().getHash());
+            resetHandler.addParameters(remoteTrackingBranchOption.get().getPointedCommit().getHash());
           } else {
             String message = "Branch '${branchName}' doesn't have remote tracking branch, so cannot be reset";
             LOG.warn(message);
             VcsNotifier.getInstance(project).notifyWarning(VCS_LOGGER_TITLE, message);
           }
 
-          handler.endOptions();
-
-          // Checking out given branch
-          GitBranchUiHandlerImpl uiHandler = new GitBranchUiHandlerImpl(project, Git.getInstance(), indicator);
-          new GitBranchWorker(project, Git.getInstance(), uiHandler)
-              .checkout(branchName, /* detach */ false, java.util.List.of(gitRepository));
+          resetHandler.endOptions();
+
+          // Check if branch to reset is not current branch - if isn't then checkout
+          Option<String> currentBranchOption = getCurrentBranchNameIfManaged(anActionEvent);
+          if (currentBranchOption.isEmpty() || !currentBranchOption.get().equals(branchName)) {
+            // Checking out given branch
+            GitBranchUiHandlerImpl uiHandler = new GitBranchUiHandlerImpl(project, Git.getInstance(), indicator);
+            new GitBranchWorker(project, Git.getInstance(), uiHandler)
+                .checkout(branchName, /* detach */ false, List.of(gitRepository));
+
+            // Check again if we are in branch to reset to be sure that checkout was successful
+            // This time we are using git4idea because GitMacheteRepository is immutable and it would return previous branch
+            GitLocalBranch nullableLocalBranch = gitRepository.getCurrentBranch();
+            if (nullableLocalBranch == null || !nullableLocalBranch.getName().equals(branchName)) {
+              LOG.error("Checkout to branch ${branchName} before reset was unsuccessful");
+              VcsNotifier.getInstance(project).notifyError(VCS_LOGGER_TITLE,
+                  "Error occurred during checking out to branch that was to be reset! Operation aborted.");
```

Comment:
Nit:
```suggestion
                  "Error occurred during checkout of the branch that was to be reset! Operation aborted.");
```

`-ing -ing` is generally considered to sound weird/poorly in English AFAIK